15314f1ddb882700dc9fabc5ca961943/update/sys_script_include_b630a09ddb9020d01206fd641d9619df.xml

```diff
@@ -718,20 +718,355 @@ IllumioUtils.prototype = {
+				if (!responseDict["status"]) {
+					return matchFound;
+				}
+				response = responseDict["response"];
+				response_status = responseDict["statusCode"];
+
+				if (response_status >= 200 && response_status < 300) {
+					response = JSON.parse(response.getBody());
+
+					// Iterate over each dimension and intialize record, if not already present, and update it.
+					for (var i = 0; i < response.length; i++) {
+						this.createOrUpdateLabelDimensionRecord(response[i], pceConfig);
+
+						if (response[i]["key"] == key) {
+							matchFound = true;
+							gs.debug("A label dimension with key " + key + " is already present on Illumio PCE for PCE configuration " + pceConfig.getValue("name") + ". Replacing the label dimension details with the ones obtained from Illumio PCE.");
+						}
+					}
+				} else {
+					gs.error("Error while validating the creation of label dimension with key - " + key + " for the PCE configuration " + pceConfig.getValue("name") + ". Status code - " + response_status + ". Response - " + JSON.stringify(response));
+					return false;
+				}
+			}
+		} catch (exc) {
+			gs.error("Error while validating the creation of label dimension with key - " + key + ". Error - " + exc);
+			return false;
+		}
+		return matchFound;
+	},
+	
+	/**
+	 * Retry failed label dimension creation
+	 * When the label dimensions creation fails on PCE, this method is to be called with the next scheduled script execution while fetching the label dimensions.
+	 * This method would retry creating the label dimensions aaded on ServiceNow onto the coorelating Illumio PCE tenant.
+	 */
+	retryFailedLabelDimensionCreation: function () {
+		var labelDimensionGr = new GlideRecord("x_illu2_illumio_illumio_label_dimension");
+		labelDimensionGr.addNullQuery("href");
+		labelDimensionGr.query();
+		while (labelDimensionGr.next()) {
+			try {
+				var details = {
+					"key": labelDimensionGr.getValue("key"),
+					"display_name_singular": labelDimensionGr.getValue("display_name_singular"),
+					"display_name_plural": labelDimensionGr.getValue("display_name_plural"),
+					"label_type_initial": labelDimensionGr.getValue("label_type_initial"),
+					"illumio_pce_configuration": labelDimensionGr.getValue("illumio_pce_configuration")
+				};
+				
+				gs.debug("Trying to create label dimension " + labelDimensionGr.getValue("key") + " on Illumio PCE for PCE configuration " + labelDimensionGr.getValue("illumio_pce_configuration") + ".");
+
+				var responseDict = this.createLabelDimensionsInPCE(details);
+				if (responseDict["status"] && !gs.nil(responseDict["href"])) {
+					labelDimensionGr.setValue("href", responseDict["href"]);
+				}
+				
+				// Mark the record to not be deleted
+				labelDimensionGr.setValue("isdeleted", false);
+				labelDimensionGr.setWorkflow(false);
+				labelDimensionGr.update();
+			} catch (exc) {
+				gs.error("Error while retrying to create the label dimension " + labelDimensionGr.getValue("key") + ". Error - " + exc);
```

Comment:
Suppose upon retry, the record was not created, what happens next? how to reprocess it in next try?

Reply:
The label dimension records would persist in ServiceNow and would be retried daily along with the execution of scheduled script that syncs the label dimensions between Illumio and ServiceNow.